File: src/Application Specific/Classes/Entity Collections/CustomerCharge/CustomerChargeSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace CustomerCharges

        Public Class CustomerChargeSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"

            Public Sub Delete(ByVal CustomerChargeID As Integer)
                _database.ClearParameter()
                _database.AddParameter("@CustomerChargeID", CustomerChargeID, True)
                _database.ExecuteSP_NO_Return("CustomerCharge_Delete")
            End Sub

            Public Function [CustomerCharge_Get](ByVal CustomerChargeID As Integer) As CustomerCharge
                _database.ClearParameter()
                _database.AddParameter("@CustomerChargeID", CustomerChargeID)

                'Get the datatable from the database store in dt
                Dim dt As DataTable = _database.ExecuteSP_DataTable("CustomerCharge_Get")

                If Not dt Is Nothing Then
                    If dt.Rows.Count > 0 Then
                        Dim oCustomerCharge As New CustomerCharge
                        With oCustomerCharge
                            .IgnoreExceptionsOnSetMethods = True
                            .SetCustomerChargeID = dt.Rows(0).Item("CustomerChargeID")
                            .SetCustomerID = dt.Rows(0).Item("CustomerID")
                            .SetMileageRate = dt.Rows(0).Item("MileageRate")
                            .SetPartsMarkup = dt.Rows(0).Item("PartsMarkup")
                            .SetRatesMarkup = dt.Rows(0).Item("RatesMarkup")

                            .SetDeleted = dt.Rows(0).Item("Deleted")
                        End With
                        oCustomerCharge.Exists = True
                        Return oCustomerCharge
                    Else
                        Return Nothing
                    End If
                Else
                    Return Nothing
                End If
            End Function

            Public Function [CustomerCharge_GetForCustomer](ByVal CustomerID As Integer) As CustomerCharge
                _database.ClearParameter()
                _database.AddParameter("@CustomerID", CustomerID)

                'Get the datatable from the database store in dt
                Dim dt As DataTable = _database.ExecuteSP_DataTable("CustomerCharge_GetForCustomer")

                If Not dt Is Nothing Then
                    If dt.Rows.Count > 0 Then
                        Dim oCustomerCharge As New CustomerCharge
                        With oCustomerCharge
                            .IgnoreExceptionsOnSetMethods = True
                            .SetCustomerChargeID = dt.Rows(0).Item("CustomerChargeID")
                            .SetCustomerID = dt.Rows(0).Item("CustomerID")
                            .SetMileageRate = dt.Rows(0).Item("MileageRate")
                            .SetPartsMarkup = dt.Rows(0).Item("PartsMarkup")
                            .SetRatesMarkup = dt.Rows(0).Item("RatesMarkup")
                            .SetDeleted = dt.Rows(0).Item("Deleted")
                        End With
                        oCustomerCharge.Exists = True
                        Return oCustomerCharge
                    Else
                        Return Nothing
                    End If
                Else
                    Return Nothing
                End If
            End Function

            Public Function [CustomerCharge_GetForSite](ByVal siteId As Integer) As CustomerCharge
                _database.ClearParameter()
                _database.AddParameter("@SiteID", siteId)

                'Get the datatable from the database store in dt
                Dim dt As DataTable = _database.ExecuteSP_DataTable("CustomerCharge_GetForSite")

                If Not dt Is Nothing Then
                    If dt.Rows.Count > 0 Then
                        Dim oCustomerCharge As New CustomerCharge
                        With oCustomerCharge
                            .IgnoreExceptionsOnSetMethods = True
                            .SetCustomerChargeID = dt.Rows(0).Item("CustomerChargeID")
                            .SetCustomerID = dt.Rows(0).Item("CustomerID")
                            .SetMileageRate = dt.Rows(0).Item("MileageRate")
                            .SetPartsMarkup = dt.Rows(0).Item("PartsMarkup")
                            .SetRatesMarkup = dt.Rows(0).Item("RatesMarkup")
                            .SetDeleted = dt.Rows(0).Item("Deleted")
                        End With
                        oCustomerCharge.Exists = True
                        Return oCustomerCharge
                    Else
                        Return Nothing
                    End If
                Else
                    Return Nothing
                End If
            End Function

            Public Function [CustomerCharge_GetAll]() As DataView
                _database.ClearParameter()
                Dim dt As DataTable = _database.ExecuteSP_DataTable("CustomerCharge_GetAll")
                dt.TableName = Entity.Sys.Enums.TableNames.tblCustomerCharge.ToString
                Return New DataView(dt)
            End Function

            Public Function [Insert](ByVal oCustomerCharge As CustomerCharge) As CustomerCharge
                _database.ClearParameter()
                AddCustomerChargeParametersToCommand(oCustomerCharge)

                oCustomerCharge.SetCustomerChargeID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("CustomerCharge_Insert"))
                oCustomerCharge.Exists = True
                Return oCustomerCharge
            End Function

            Public Sub [Update](ByVal oCustomerCharge As CustomerCharge)
                _database.ClearParameter()
                AddCustomerChargeParametersToCommand(oCustomerCharge)
                _database.ExecuteSP_NO_Return("CustomerCharge_Update")
            End Sub

            Public Sub [UpdateALL](ByVal MileageRate As Double, ByVal PartsMarkup As Double, ByVal RatesMarkup As Double)
                _database.ClearParameter()
                _database.AddParameter("@MileageRate", MileageRate, True)
                _database.AddParameter("@PartsMarkup", PartsMarkup, True)
                _database.AddParameter("@RatesMarkup", RatesMarkup, True)
                _database.ExecuteSP_NO_Return("CustomerCharge_UpdateALL")
            End Sub

            Private Sub AddCustomerChargeParametersToCommand(ByRef oCustomerCharge As CustomerCharge)
                With _database
                    .AddParameter("@CustomerID", oCustomerCharge.CustomerID, True)
                    .AddParameter("@MileageRate", oCustomerCharge.MileageRate, True)
                    .AddParameter("@PartsMarkup", oCustomerCharge.PartsMarkup, True)
                    .AddParameter("@RatesMarkup", oCustomerCharge.RatesMarkup, True)
                End With
            End Sub

#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/CustomerCharge/CustomerChargeSQL.cs
using System.Data;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace CustomerCharges
    {
        public class CustomerChargeSQL
        {
            private Sys.Database _database;

            public CustomerChargeSQL(Sys.Database database)
            {
                _database = database;
            }

            
            public void Delete(int CustomerChargeID)
            {
                _database.ClearParameter();
                _database.AddParameter("@CustomerChargeID", CustomerChargeID, true);
                _database.ExecuteSP_NO_Return("CustomerCharge_Delete");
            }

            public CustomerCharge CustomerCharge_Get(int CustomerChargeID)
            {
                _database.ClearParameter();
                _database.AddParameter("@CustomerChargeID", CustomerChargeID);

                // Get the datatable from the database store in dt
                var dt = _database.ExecuteSP_DataTable("CustomerCharge_Get");
                if (dt is object)
                {
                    if (dt.Rows.Count > 0)
                    {
                        var oCustomerCharge = new CustomerCharge();
                        oCustomerCharge.IgnoreExceptionsOnSetMethods = true;
                        oCustomerCharge.SetCustomerChargeID = dt.Rows[0]["CustomerChargeID"];
                        oCustomerCharge.SetCustomerID = dt.Rows[0]["CustomerID"];
                        oCustomerCharge.SetMileageRate = dt.Rows[0]["MileageRate"];
                        oCustomerCharge.SetPartsMarkup = dt.Rows[0]["PartsMarkup"];
                        oCustomerCharge.SetRatesMarkup = dt.Rows[0]["RatesMarkup"];
                        oCustomerCharge.SetDeleted = Conversions.ToBoolean(dt.Rows[0]["Deleted"]);
                        oCustomerCharge.Exists = true;
                        return oCustomerCharge;
                    }
                    else
                    {
                        return null;
                    }
                }
                else
                {
                    return null;
                }
            }

            public CustomerCharge CustomerCharge_GetForCustomer(int CustomerID)
            {
                _database.ClearParameter();
                _database.AddParameter("@CustomerID", CustomerID);

                // Get the datatable from the database store in dt
                var dt = _database.ExecuteSP_DataTable("CustomerCharge_GetForCustomer");
                if (dt is object)
                {
                    if (dt.Rows.Count > 0)
                    {
                        var oCustomerCharge = new CustomerCharge();
                        oCustomerCharge.IgnoreExceptionsOnSetMethods = true;
                        oCustomerCharge.SetCustomerChargeID = dt.Rows[0]["CustomerChargeID"];
                        oCustomerCharge.SetCustomerID = dt.Rows[0]["CustomerID"];
                        oCustomerCharge.SetMileageRate = dt.Rows[0]["MileageRate"];
                        oCustomerCharge.SetPartsMarkup = dt.Rows[0]["PartsMarkup"];
                        oCustomerCharge.SetRatesMarkup = dt.Rows[0]["RatesMarkup"];
                        oCustomerCharge.SetDeleted = Conversions.ToBoolean(dt.Rows[0]["Deleted"]);
                        oCustomerCharge.Exists = true;
                        return oCustomerCharge;
                    }
                    else
                    {
                        return null;
                    }
                }
                else
                {
                    return null;
                }
            }

            public CustomerCharge CustomerCharge_GetForSite(int siteId)
            {
                _database.ClearParameter();
                _database.AddParameter("@SiteID", siteId);

                // Get the datatable from the database store in dt
                var dt = _database.ExecuteSP_DataTable("CustomerCharge_GetForSite");
                if (dt is object)
                {
                    if (dt.Rows.Count > 0)
                    {
                        var oCustomerCharge = new CustomerCharge();
                        oCustomerCharge.IgnoreExceptionsOnSetMethods = true;
                        oCustomerCharge.SetCustomerChargeID = dt.Rows[0]["CustomerChargeID"];
                        oCustomerCharge.SetCustomerID = dt.Rows[0]["CustomerID"];
                        oCustomerCharge.SetMileageRate = dt.Rows[0]["MileageRate"];
                        oCustomerCharge.SetPartsMarkup = dt.Rows[0]["PartsMarkup"];
                        oCustomerCharge.SetRatesMarkup = dt.Rows[0]["RatesMarkup"];
                        oCustomerCharge.SetDeleted = Conversions.ToBoolean(dt.Rows[0]["Deleted"]);
                        oCustomerCharge.Exists = true;
                        return oCustomerCharge;
                    }
                    else
                    {
                        return null;
                    }
                }
                else
                {
                    return null;
                }
            }

            public DataView CustomerCharge_GetAll()
            {
                _database.ClearParameter();
                var dt = _database.ExecuteSP_DataTable("CustomerCharge_GetAll");
                dt.TableName = Sys.Enums.TableNames.tblCustomerCharge.ToString();
                return new DataView(dt);
            }

            public CustomerCharge Insert(CustomerCharge oCustomerCharge)
            {
                _database.ClearParameter();
                AddCustomerChargeParametersToCommand(ref oCustomerCharge);
                oCustomerCharge.SetCustomerChargeID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("CustomerCharge_Insert"));
                oCustomerCharge.Exists = true;
                return oCustomerCharge;
            }

            public void Update(CustomerCharge oCustomerCharge)
            {
                _database.ClearParameter();
                AddCustomerChargeParametersToCommand(ref oCustomerCharge);
                _database.ExecuteSP_NO_Return("CustomerCharge_Update");
            }

            public void UpdateALL(double MileageRate, double PartsMarkup, double RatesMarkup)
            {
                _database.ClearParameter();
                _database.AddParameter("@MileageRate", MileageRate, true);
                _database.AddParameter("@PartsMarkup", PartsMarkup, true);
                _database.AddParameter("@RatesMarkup", RatesMarkup, true);
                _database.ExecuteSP_NO_Return("CustomerCharge_UpdateALL");
            }

            private void AddCustomerChargeParametersToCommand(ref CustomerCharge oCustomerCharge)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@CustomerID", oCustomerCharge.CustomerID, true);
                    withBlock.AddParameter("@MileageRate", oCustomerCharge.MileageRate, true);
                    withBlock.AddParameter("@PartsMarkup", oCustomerCharge.PartsMarkup, true);
                    withBlock.AddParameter("@RatesMarkup", oCustomerCharge.RatesMarkup, true);
                }
            }

            
        }
    }
}